msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -2058,6 +2078,22 @@ boolean matches(@NonNull final String localAccountId,
 
                 if (null != tenantProfiles && !tenantProfiles.isEmpty()) {
                     for (final Map.Entry<String, ITenantProfile> profileEntry : tenantProfiles.entrySet()) {
+                        boolean isEmpty = TextUtils.isEmpty(profileEntry.getValue().getId());
+                        if(isEmpty){
+                            com.microsoft.identity.common.logging.Logger.info(
+                                    TAG + methodName,
+                                    "For localAccountId identifier: " + localAccountId +
+                                            " profileEntry getId() is empty."
+                            );
+                        } else {
+                            com.microsoft.identity.common.logging.Logger.info(
+                                    TAG + methodName,
+                                    "For localAccountId identifier: " + localAccountId +
+                                            " and profileEntry account id :" + profileEntry.getValue().getId() +
+                                            ", match : " + localAccountId.contains(profileEntry.getValue().getId())
+                            );
+                        }
+
                         if (!TextUtils.isEmpty(profileEntry.getValue().getId()) &&
                                 localAccountId.contains(profileEntry.getValue().getId())) {
                             return true;
```

Comment:
Down there on line 2105, I'd like us to log the fact that we are returning false with no match.